test_Gather fails randomly

=================================== FAILURES ===================================
____________________ TestGatherElements.testGatherElements1 ____________________
self = <tests.tests_inference.tests_layer.test_c_python.test_GatherElements.TestGatherElements testMethod=testGatherElements1>
    def testGatherElements1(self):
        in_shape = list(np.random.randint(1, 100, size=3))
        in_shape = (1, int(in_shape[0]), int(in_shape[1]), int(in_shape[2]))
>       out_shape = (1, int(np.random.randint(1, in_shape[1])), int(in_shape[2]), int(in_shape[3]))
tests/tests_inference/tests_layer/test_c_python/test_GatherElements.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mtrand.pyx:748: in numpy.random.mtrand.RandomState.randint
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
>   ???
E   ValueError: low >= high
_bounded_integers.pyx:1247: ValueError
=============================== warnings summary ===============================
../../../../../opt/hostedtoolcache/Python/3.10.18/x64/lib/python3.10/site-packages/tensorflow/lite/python/util.py:52
../../../../../opt/hostedtoolcache/Python/3.10.18/x64/lib/python3.10/site-packages/tensorflow/lite/python/util.py:52
  /opt/hostedtoolcache/Python/3.10.18/x64/lib/python3.10/site-packages/tensorflow/lite/python/util.py:52: DeprecationWarning: jax.xla_computation is deprecated. Please use the AOT APIs.
    from jax import xla_computation as _xla_computation
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/tests_inference/tests_layer/test_c_python/test_GatherElements.py::TestGatherElements::testGatherElements1 - ValueError: low >= high
====== 1 failed, 408 passed, 14 xfailed, 2 warnings in 281.08s (0:04:41) =======
Error: Process completed with exit code 1.

--- a/tests/tests_inference/tests_layer/test_c_python/test_GatherElements.py
+++ b/tests/tests_inference/tests_layer/test_c_python/test_GatherElements.py
@@ -35,7 +35,7 @@ class TestGatherElements(acetoneTestCase.AcetoneTestCase):
     def testGatherElements1(self):
         in_shape = list(np.random.randint(1, 100, size=3))
         in_shape = (1, int(in_shape[0]), int(in_shape[1]), int(in_shape[2]))
-        out_shape = (1, int(np.random.randint(1, in_shape[1])), int(in_shape[2]), int(in_shape[3]))
+        out_shape = (1, int(np.random.randint(1, 1+in_shape[1])), int(in_shape[2]), int(in_shape[3]))
 
         model_input_name = "X"
         X = onnx.helper.make_tensor_value_info(model_input_name,
@@ -78,7 +78,7 @@ class TestGatherElements(acetoneTestCase.AcetoneTestCase):
     def testGatherElements2(self):
         in_shape = list(np.random.randint(1, 100, size=3))
         in_shape = (1, int(in_shape[0]), int(in_shape[1]), int(in_shape[2]))
-        out_shape = (1, int(in_shape[1]), int(np.random.randint(1, in_shape[2])), int(in_shape[3]))
+        out_shape = (1, int(in_shape[1]), int(np.random.randint(1, 1+in_shape[2])), int(in_shape[3]))
 
         model_input_name = "X"
         X = onnx.helper.make_tensor_value_info(model_input_name,
@@ -121,7 +121,7 @@ class TestGatherElements(acetoneTestCase.AcetoneTestCase):
     def testGatherElements3(self):
         in_shape = list(np.random.randint(1, 100, size=3))
         in_shape = (1, int(in_shape[0]), int(in_shape[1]), int(in_shape[2]))
-        out_shape = (1, int(in_shape[1]), int(in_shape[2]), int(np.random.randint(1, in_shape[3])))
+        out_shape = (1, int(in_shape[1]), int(in_shape[2]), int(np.random.randint(1, 1+in_shape[3])))
 
         model_input_name = "X"
         X = onnx.helper.make_tensor_value_info(model_input_name,

--- a/tests/tests_inference/tests_layer/test_c_reference/test_GatherElements.py
+++ b/tests/tests_inference/tests_layer/test_c_reference/test_GatherElements.py
@@ -30,7 +30,7 @@ class TestGatherElements(acetoneTestCase.AcetoneTestCase):
     def testGatherElements1(self):
         in_shape = list(np.random.randint(1, 100, size=3))
         in_shape = (1, int(in_shape[0]), int(in_shape[1]), int(in_shape[2]))
-        out_shape = (1, int(np.random.randint(1, high=in_shape[1])), int(in_shape[2]), int(in_shape[3]))
+        out_shape = (1, int(np.random.randint(1, high=1+in_shape[1])), int(in_shape[2]), int(in_shape[3]))
 
         model_input_name = "X"
         X = onnx.helper.make_tensor_value_info(model_input_name,
@@ -78,7 +78,7 @@ class TestGatherElements(acetoneTestCase.AcetoneTestCase):
     def testGatherElements2(self):
         in_shape = list(np.random.randint(1, 100, size=3))
         in_shape = (1, int(in_shape[0]), int(in_shape[1]), int(in_shape[2]))
-        out_shape = (1, int(in_shape[1]), int(np.random.randint(1, in_shape[2])), int(in_shape[3]))
+        out_shape = (1, int(in_shape[1]), int(np.random.randint(1, 1+in_shape[2])), int(in_shape[3]))
 
         model_input_name = "X"
         X = onnx.helper.make_tensor_value_info(model_input_name,
@@ -126,7 +126,7 @@ class TestGatherElements(acetoneTestCase.AcetoneTestCase):
     def testGatherElements3(self):
         in_shape = list(np.random.randint(1, 100, size=3))
         in_shape = (1, int(in_shape[0]), int(in_shape[1]), int(in_shape[2]))
-        out_shape = (1, int(in_shape[1]), int(in_shape[2]), int(np.random.randint(1, in_shape[3])))
+        out_shape = (1, int(in_shape[1]), int(in_shape[2]), int(np.random.randint(1, 1+in_shape[3])))
 
         model_input_name = "X"
         X = onnx.helper.make_tensor_value_info(model_input_name,

--- a/tests/tests_inference/tests_layer/test_python_reference/test_GatherElements.py
+++ b/tests/tests_inference/tests_layer/test_python_reference/test_GatherElements.py
@@ -32,7 +32,7 @@ class TestGatherElements(acetoneTestCase.AcetoneTestCase):
         in_shape = (1, int(in_shape[0]), int(in_shape[1]), int(in_shape[2]))
         out_shape = (
             1,
-            int(np.random.randint(1, in_shape[1])),
+            int(np.random.randint(1, 1+in_shape[1])),
             int(in_shape[2]),
             int(in_shape[3]),
         )
@@ -96,7 +96,7 @@ class TestGatherElements(acetoneTestCase.AcetoneTestCase):
         out_shape = (
             1,
             int(in_shape[1]),
-            int(np.random.randint(1, in_shape[2])),
+            int(np.random.randint(1, 1+in_shape[2])),
             int(in_shape[3]),
         )
 
@@ -160,7 +160,7 @@ class TestGatherElements(acetoneTestCase.AcetoneTestCase):
             1,
             int(in_shape[1]),
             int(in_shape[2]),
-            int(np.random.randint(1, in_shape[3])),
+            int(np.random.randint(1, 1+in_shape[3])),
         )
 
         model_input_name = "X"

